crates/cubecl-linalg/src/tensor/contiguous.rs

```diff
@@ -61,11 +74,41 @@ pub fn into_contiguous<R: Runtime, E: CubePrimitive>(
         input.strides,
         rank - 1,
     );
+    let num_vecs = num_elems / vectorization_factor as usize;
+    let approx_sm = 64;
+    let approx_simul_vecs = approx_sm * CubeDim::default().num_elems();
+    let elems_per_unit = match num_vecs as u32 / approx_simul_vecs {
+        0..2 => 1,
+        2..4 => 2,
+        4..8 => 4,
+        8.. => 8,
+    };
+
+    // TODO: Benchmark to find good default prefetch, for now preserve existing behaviour
+    into_contiguous_prefetch(client, input, elems_per_unit)
+}
+
+/// Make a jit tensor contiguous.
+pub fn into_contiguous_prefetch<R: Runtime, E: CubePrimitive>(
+    client: &ComputeClient<R::Server, R::Channel>,
+    input: TensorHandleRef<'_, R>,
+    elems_per_unit: u32,
```

Comment:
The vectorization factor could be passed as input, we're already calculating it in the `into_contiguous` function. Also is there a reason why this function is public?

Reply:
To make it possible to override the `elems_per_unit` based on autotuning or other heuristics that may more more specific to that use case. That's why I don't want to pass the vectorization: the function is supposed to be simple to use from things like `burn` so it can be tuned (auto or not) easily.